src/middlewared/middlewared/etc_files/local/ssh/config.py

```diff
@@ -24,6 +25,16 @@ def generate_ssh_config(middleware):
         if os.path.exists(os.path.join(SSH_CONFIG_PATH, f)):
             os.chmod(os.path.join(SSH_CONFIG_PATH, f), 0o600)
 
+    twofactor_auth = middleware.call_sync('auth.twofactor.config')
+    if twofactor_auth['enabled'] and twofactor_auth['services']['ssh']:
+        with open(USERS_OATH_FILE, 'w') as f:
+            f.write(
+                f'HOTP/T{twofactor_auth["interval"]}/{twofactor_auth["otp_digits"]}\troot\t-\t'
+                f'{base64.b16encode(base64.b32decode(twofactor_auth["secret"])).decode()}\n'
+            )
+
+        os.chmod(USERS_OATH_FILE, 0o600)
+
```

Comment:
What if it gets disabled? I think we should remove the file, so we are back at square one :P